ONSdigital/eq-author-app
# Ear 1317 modal for importing questions

### What is the context of this PR?

#### Components + business logic (@MattElcock)

This PR builds the question picker modal necessary for the Importing process.

* Renamed `QuestionPicker` to `AnswerPicker`
* * This is the modal used on the Calculated Summary page. It was incorrectly named as it selects answers, not questions.
* Moved `AnswerPicker` into `src/components`
* * It's a component within it's own right.
* Built `QuestionPicker` within `src/components`
* Added an example implementation of the above to Storybook
* Rejigged the modals used within the importing process to be in their own folder in Storybook, under patterns.

#### Tests (@ashtuck)

Tests for the`Question Picker Component`. These tests include - 

Render Test
Function Firing Tests
Interaction Tests
Search Bar Tests

Also fixed a bug in the `Item Component` in the same directory. Was a bit of a weird one, all I’ve done is change the prop so it’s a template literal. Think it might be a bug with `Styled Components`, not sure though. Documented in `Styled Component Bug` commit.

### How to review

Boot up Storybook and look under `Patterns/Importing content/Questions` and try out the modal.

Make sure the Calculated Summary content picker still works.

Run `yarn test` and search for `QuestionPicker`. Check these have passed.
I’ve run through and checked for false positives and it’s looking good, but if you want to check for that as well to be sure.

- Import the **Capability examples** questionnaire from pre-prod Author into your local environment and:
  - ensure it can be opened in Author;
  - then, ensure it can be viewed in Runner by pressing the **view survey** button
- Does this require a migration? We need one if existing JSON schema properties change

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
